homework9/Task1.py

```diff
@@ -0,0 +1,26 @@
+"""
+Write a function that merges integer from sorted files and returns an iterator
+file1.txt:
+1
+3
+5
+file2.txt:
+2
+4
+6
+>>> list(merge_sorted_files(["file1.txt", "file2.txt"]))
+[1, 2, 3, 4, 5, 6]
+"""
+from contextlib import ExitStack
+from itertools import zip_longest
+from pathlib import Path
+from typing import Iterator, List, Union
+
+
+def merge_sorted_files(file_list: List[Union[Path, str]]) -> Iterator:
+    with ExitStack() as stack:
+        files = [stack.enter_context(open(file)) for file in file_list]
+        numbers = [[num.strip("\n") for num in line] for line in files]
```

Comment:
You load all the data into a memory here, which is not optimal. What if there are 5 files 10 Gb each?